blob/gcsblob/gcsblob.go

```diff
@@ -65,11 +68,18 @@ func (r *reader) Attrs() *driver.ObjectAttrs {
 func (b *bucket) NewRangeReader(ctx context.Context, key string, offset, length int64) (driver.Reader, error) {
 	bkt := b.client.Bucket(b.name)
 	obj := bkt.Object(key)
+	attrs, err := obj.Attrs(ctx)
```

Comment:
It's unfortunate that this requires a second network call, but it was bound to happen eventually.

My one concern is that there is a race here: `Attrs` could be for a different version than the reader is for. In this case, there's not that much cause for concern, but I'd like to add the constraint now so that as the code evolves, we don't hit the issue later.

Reply:
One alternative is to not call `obj.Attrs` inside `NewRangeReader`, instead, call it when user explicitly asks for it the first time. At that point we can check whether this field is non-zero and decide to make the call. WDYT?

Reply:
The [updated field](https://godoc.org/google.golang.org/api/storage/v1#Object.Updated) is included in the object but not carried by the client library we use here. I suggest removing the implementation here for GCS and keep those for S3 and file, add comments to the blob package saying that this ModTime field is optional. We'll try to have it supported by the client library first and add it back here.